roborock/device_features.py

```diff
@@ -174,6 +175,51 @@ class ProductFeatures(StrEnum):
 }
 
 
+def update_features():
+    """Update the feature mapping."""
+    # Their firmware describes what doesn't have these features - not which one does.
+    # Matching that for ease of future updates.
+    PRODUCTS_WITHOUT_CUSTOM_CLEAN = {
+        RoborockProductNickname.TANOS,
+        RoborockProductNickname.RUBYPLUS,
+        RoborockProductNickname.RUBYSC,
+        RoborockProductNickname.RUBYSE,
+    }
+    PRODUCTS_WITHOUT_DEFAULT_3D_MAP = {
+        RoborockProductNickname.TANOS,
+        RoborockProductNickname.TANOSSPLUS,
+        RoborockProductNickname.TANOSE,
+        RoborockProductNickname.TANOSV,
+        RoborockProductNickname.RUBYPLUS,
+        RoborockProductNickname.RUBYSC,
+        RoborockProductNickname.RUBYSE,
+    }
+    PRODUCTS_WITHOUT_PURE_CLEAN_MOP = {
+        RoborockProductNickname.TANOS,
+        RoborockProductNickname.TANOSE,
+        RoborockProductNickname.TANOSV,
+        RoborockProductNickname.TANOSSLITE,
+        RoborockProductNickname.TANOSSE,
+        RoborockProductNickname.TANOSSC,
+        RoborockProductNickname.ULTRONLITE,
+        RoborockProductNickname.ULTRONE,
+        RoborockProductNickname.RUBYPLUS,
+        RoborockProductNickname.RUBYSLITE,
+        RoborockProductNickname.RUBYSC,
+        RoborockProductNickname.RUBYSE,
+    }
+    for product, feature_list in product_feature_map.items():
```

Comment:
Don't reall love updating the map like this. Can we use the logic here to build the entire map from the start?

Reply:
I'm fine with a dynamic feature map, i'm just saying this mutating a global variatble python object `product_feature_map`.  Create the `product_feature_map` right from the start rather than calling a method to update this global variable.